### ISSUE TITLE
Try/Catch Question

### ISSUE THREAD

**Issue by psasser** [woocommerce/wc-api-php#64]
Does this code....

$woocommerce = new Client(
	$storeurl, 
	$consumerkey, 
	$consumersecret,
	[
		'wp_api' => true,
		'version' => 'wc/v1',
	]
);

Need to be within the try/catch.  Or do we put the try/catch below that when we are creating/update a product? 

I'm trying to determine why I'm getting these errors.  http://pastebin.com/SqiPJLTa.  I saw on an issue someone else posted you said because it isn't between a try/catch.  So I'm seeing if that would resolve it.  Strangely I have product being created via REST API however sometime it doesn't work and sometimes it does.  The cron to create runs every 5 mins.  Sometimes I get an error on a product when creating it and then 5 min later it goes on the site without issue.  Any thoughts as to why that might be occuring??

**Comment by claudiosanches** [comment]
If you have cURL installed, the `new Clinet()` should not throw any Exception and you can onlu use for the get/post/put/delete methods.

**Comment by psasser** [comment]
phpinfo() says cURL support is enabled.  Any ideas as to why I'm still throwing an exception?  I would think that it could be the data that I'm passing to the  in the product array

$newproduct = $woocommerce->post('products', $newproductarray);

However sometimes this works and sometimes it doesn't.

**Comment by psasser** [comment]
I stumbled on this comment 

https://wordpress.org/support/topic/cannot-getset-product-information-through-api/

and this guy says to remove the version from the options.  Would this have any impact on it?

**Comment by psasser** [comment]
Also I just realized you made some commits 21 days ago and I don't have those.  Could that be what's causing issues??

**Comment by claudiosanches** [comment]
@psasser because he used a wrong version number, since he is not using the new WP-API integration.
In your case you are doing correct.

But as I said, you should use

```php
use Automattic\WooCommerce\HttpClient\HttpClientException;

try {
    $newproduct = $woocommerce->post('products', $newproductarray);
} catch(HttpClientException $e) {
    \print_r($e->getMessage() . PHP_EOL);
}
```

The error message that you are saying about the `Automattic\WooCommerce\HttpClient\HttpClientException: Syntax error in /websites/acosta/wp-content/plugins/simple-consign-woocommerce-tie-in/vendor/automattic/woocommerce/src/WooCommerce/HttpClient/HttpClient.php:331` is because have something wrong with the JSON response, in this case not a valid JSON response, since is returning HTML probably because a PHP warning with the JSON.

**Comment by psasser** [comment]
Thanks for the advice.  I went ahead and added the code like you suggested above, and I also updated my code to your latest GitHub release and it works to a certain extent.  

Essentially if I hit the request in the browser by just typing in the web address to the page all works as expected.  However if I call the same code from a PHP cron job I keep getting the same exception...

Automattic\WooCommerce\HttpClient\HttpClientException: Syntax error in /websites/acosta/wp-content/plugins/simple-consign-woocommerce-tie-in/vendor/automattic/woocommerce/src/WooCommerce/HttpClient/HttpClient.php:332

I have no idea how to troubleshoot this.  Any thoughts? I see what you're saying about invalid JSON being returned because of PHP warning is there anyway I can find out what PHP warning is being returned?  Should I just suppress PHP warnings on the site or something?? Or do you have any ideas how I can fix??

Thanks.

**Comment by claudiosanches** [comment]
Need to check the log on your site, since something on your WordPress/WooCommerce installation is throwing a PHP warning message.
So it's the same steps that debug anything on WordPress.

**Comment by psasser** [comment]
For anyone else having any issues with this the problem resulted in how this URL was added to my code cron job...

http://www.example.com/wp-json/wc/v1/products

I was using get_site_url() to create the $woocommerce client.  Anyway when I ran get_site_url() in the browser it resulted in https://www.example.com (which is what I needed), but when my cron job ran it, it resulted in http://www.example.com.  So long store short I changed get_site_url() to be get_site_url('', '', 'https');  Wasted countless hours on a 302 error.
